### PR TITLE
WIP: Documentation build process

### PR DESCRIPTION
**Pull request by minrwhite**
# Do not merge

Using the setup linked to in a comment by @ThomasWaldmann on #660 as a base, this PR allows local building of the documentation in html format, (pdf)latex and epub formats as well as checking of external links. This requires that GNU Make and `python` are both installed locally, as well as the `pip` packages `Sphinx` and `sphinx_rtd_theme`. Navigate to /docs and run `make help` to see a list of valid targets.

### CODE REVIEW

**Comment by gr2m** [comment]
Thanks for picking it up Ben! How about we add a `README.md` file to the `docs` folder with info on what this folder is and how to build the docs locally? If it makes any difference, I don’t think we’ll need any other build formats than html locally.

**Comment by ThomasWaldmann** [comment]
If you consider facts about your development/documentation process as documentation also, you could just have a "development" section there to put such stuff.

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ node_modules
 .DS_Store
 .hoodie
 .nyc_output
+docs/_build/**/*
```
Comment (gr2m) [inline .gitignore:7]: I _think_ `docs/_build` should be enough?

**Comment by fozy81** [comment]
Hi, I've created a [fork](https://github.com/fozy81/hoodie), and moved all docs into a 'developers' sub-directory in the docs/ folder. 

-  I think its good to have all the docs in one place along side other getting started guides etc. Maybe this is a break with convention? [Readthedocs](http://docs.readthedocs.io/en/latest/) website has the docs organised in this way. 
- I've drafted a '[How to contribute to documentation](https://github.com/fozy81/hoodie/blob/master/docs/developers/CONTRIBUTING_DOCS.rst)', this explains how to make small changes on github or build the documents locally for bigger changes. Only a early draft...
- Also drafted a '[Documentation style guide](https://github.com/fozy81/hoodie/blob/master/docs/developers/DOCS_STYLE.rst)' which provides guidance on how best to write documentation and also lists the Travis tests. 
- Add travis tests...this got complicated...the tools are mainly built for .md files so I've used pandoc to temporarily convert all the .rst to .md
- Using awesome_bot & textlint to travis test dead links and grammar, spelling, style. There's lots more things we could test...don't want the tests to be too pedantic, most set to warnings only(?)
- I had to set `sudo: required` in travis - this slows the build time, maybe other issues with doing this?
Hopefully this will help implement https://github.com/hoodiehq/hoodie/pull/671 and https://github.com/hoodiehq/hoodie/pull/660 - let me know your thoughts, lots of potential changes - and I'm new to a lot of this! :ear: :smile:  

The build is currently [failing](https://travis-ci.org/fozy81/hoodie/builds/204012114) because of dead links... sometimes you need to break things to make them better  :worried:

**Comment by fozy81** [comment]
I've got the tests to all pass! lots of changes as described in previous comment https://github.com/hoodiehq/hoodie/pull/694

**Comment by gr2m** [comment]
@fozy81 I think we go all the relevant bits merged, right? So we can close this PR?

**Pull request closed without merging**